cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -135,11 +188,54 @@ impl InterfaceDiagnostics {
     }
 }
 
+/// Integration version of a `PartialResponse` that uses `InterfaceDiagnistics` for simpler (de)serialization
+#[cfg(feature = "partial-eval")]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct InterfaceResidualResponse {
+    /// A residual set of policies. Determining the concrete response requires further processing.
+    residuals: HashMap<PolicyId, serde_json::Value>,
+    /// Diagnostics providing more information on how this decision was reached
+    diagnostics: InterfaceDiagnostics,
+}
+
+#[cfg(feature = "partial-eval")]
+impl InterfaceResidualResponse {
+    /// Construct an `InterfaceResidualResponse`
+    pub fn new(residuals: HashMap<PolicyId, serde_json::Value>, reason: HashSet<PolicyId>, errors: HashSet<String>) -> Self {
+        Self {
+            residuals,
+            diagnostics: InterfaceDiagnostics { reason, errors },
+        }
+    }
+}
+
+#[cfg(feature = "partial-eval")]
+impl From<PartialResponse> for InterfaceResidualResponse {
+    fn from(partial_response: PartialResponse) -> Self {
+        match partial_response {
+            PartialResponse::Residual(residual) => Self::new(
+                residual.residuals().policies()
+                    .map( | policy| (policy.id().clone(), policy.to_json().unwrap()))
+                    .collect(),
+                residual.diagnostics().reason().cloned().collect(),
+                residual
+                    .diagnostics()
+                    .errors()
+                    .map(ToString::to_string)
+                    .collect()
+            ),
+            PartialResponse::Concrete(_) => panic!("Unsupported")
+        }
+    }
+}
+
 #[derive(Debug, Serialize, Deserialize)]
 #[serde(untagged)]
 enum AuthorizationAnswer {
     ParseFailed { errors: Vec<String> },
-    Success { response: InterfaceResponse },
+    Concrete { response: InterfaceResponse },
+    #[cfg(feature = "partial-eval")]
+    Residuals { response: InterfaceResidualResponse },
```

Comment:
I think we may want to have a separate struct `PartialAuthorizationAnswer` and keep this unmodified. If we turn the `parial-eval` flag on by default at some point, I think it will make more sense to have them separate.

Reply:
This enum is private to the module and only used to drive JSON serialization

Reply:
IMO this is probably fine for the JSON format.  When we document the interface we should document it as the JSON objects for each are  distinct separate types, but unifying them in the implementation is fine.

Reply:
I just changed it as per @andrewmwells-amazon 's recommendation. At the end of the day, it's easy to change and private to the module so you do whatever you want with it ;)